python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -754,6 +726,10 @@ def _flush(self, _objs=None):
             objs = []
 
         obj_ids = []
+
+        for obj in objs:
+            obj._acquireParentLock()
+
```

Comment:
I feel very uneasy doing this without a context manager. Would it be possible to acquire the lock per object, work with it and then release it? Otherwise, does the locking on the `GangaObject` not suffice? It should otherwise we have a problem.